token-lending/program/src/processor.rs

```diff
@@ -2494,7 +2526,10 @@ fn validate_switchboard_keys(
     if *switchboard_feed_info.key == spl_token_lending::NULL_PUBKEY {
         return Ok(());
     }
-    if &lending_market.switchboard_oracle_program_id != switchboard_feed_info.owner {
+    if switchboard_feed_info.owner != &lending_market.switchboard_oracle_program_id
+        && switchboard_feed_info.owner != &spl_token_lending::SWITCHBOARD_V2_MAINNET
+        && switchboard_feed_info.owner != &spl_token_lending::SWITCHBOARD_V2_DEVNET
```

Comment:
This is fine but it's more compute efficient to select the appropriate program id at compile time with a cargo feature. It would also be nice to make use of the reserve config to know which version of the switchboard program is expected.

Reply:
yeh i think for addressing this it will just be that once v1 is depericated will just put the apropriate switchboard program address on the market and check vs that